src/include/storage/metric/abstract_metric.h

```diff
@@ -0,0 +1,354 @@
+#pragma once
+
+#include <emmintrin.h>
+#include <atomic>
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "catalog/catalog_defs.h"
+#include "storage/metric/abstract_raw_data.h"
+#include "storage/metric/metric_defs.h"
+
+namespace terrier {
+
+namespace transaction {
+class TransactionContext;
+}  // namespace transaction
+
+namespace storage::metric {
```

Comment:
Should this be under the storage namespace at all? Somehow I feel like you can just have a metric namespace